compiler/lib/eval.ml

```diff
@@ -359,14 +432,21 @@ let eval_instr info ((x, loc) as i) =
                 ( x
                 , Prim
                     ( prim
-                    , List.map2 prim_args prim_args' ~f:(fun arg c ->
-                          match c with
-                          | Some ((Int _ | Float _ | NativeString _) as c) -> Pc c
-                          | Some (String _ as c) when Config.Flag.use_js_string () -> Pc c
-                          | Some _
+                    , List.map2 prim_args prim_args' ~f:(fun arg (c : constant option) ->
+                          match c, target with
+                          | Some ((Int _ | NativeString _) as c), _ -> Pc c
+                          | Some ((Int32 _ | NativeInt _) as c), `Wasm -> Pc c
```

Comment:
```suggestion
                          | Some ((Int32 _ | NativeInt _) as c), `Wasm -> arg
```

Reply:
Why not use the statically evaluated constant in Wasm?

Reply:
Wouldn't that duplicate the allocation ?

Reply:
Ah, you’re right. Done.